plugins/transport-nio/src/main/java/org/opensearch/http/nio/HttpReadWriteHandler.java

```diff
@@ -172,7 +173,7 @@ private void handleRequest(Object msg) {
         final HttpPipelinedRequest pipelinedRequest = (HttpPipelinedRequest) msg;
         boolean success = false;
         try {
-            transport.incomingRequest(pipelinedRequest, nioHttpChannel);
+            transport.incomingRequest(pipelinedRequest, nioHttpChannel, RestHandlerContext.EMPTY);
```

Comment:
This feels like we are leaking an @Internal interface to a public api with this change, can this be avoided?

Reply:
tbh I'm not quite sure how the transport-nio model comes into the picture. I think RestHandlerContext is a generic concept regardless of which transport is used as it is an object that contains the context needed for the Rest Handler

Reply:
Removed this to avoid passing this information around